components/src/main/kotlin/com/mineinabyss/geary/ecs/actions/components/ComponentActions.kt

```diff
@@ -32,11 +34,14 @@ public class AddComponentAction(
  */
 @Serializable
 @SerialName("geary:remove")
-public data class RemoveComponentAction(override val components: Set<String>) : ComponentAction() {
+public data class RemoveComponentAction(private val components: Set<String>) : GearyAction() {
+    @Transient
+    private val _components = Components(components)
```

Comment:
You don't typically don't use `_` for private fields in Java unless they're a backing field (e.x. a private mutable list with a public counterpart that returns an immutable list).

Remove in this case.